File: LeetCode/Easy/463-island-perimeter/solution.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int row = grid.length; int col = grid[0].length;
        int p = 0;
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(grid[i][j]==1){
                    if(i==0||grid[i-1][j]==0) p++;
                    if(j==0||grid[i][j-1]==0) p++;
                    if(i==row-1||grid[i+1][j]==0) p++;
                    if(j==col-1||grid[i][j+1]==0) p++;
                }
            }
        }
        return p;
    }
}

File: LeetCode/Easy/463-island-perimeter/solution.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int rows = grid.size();
        int col = grid[0].size();
        int p = 0;
        for(int i=0;i<rows;i++){
            for(int j=0;j<col;j++){
                if(grid[i][j]==1){
                    if(i==0||grid[i-1][j]==0) p++; //up
                if(j==0||grid[i][j-1]==0) p++; //left
                if(i==rows-1||grid[i+1][j]==0) p++; //down;
                if(j==col-1||grid[i][j+1]==0) p++; //right;
                }
            }
        }
        return p;
    }
};